### PR TITLE
Add coverage_report target

### PR DESCRIPTION
**Pull request by Bronek**
Reusing the same `CodeCoverage.cmake` as in https://github.com/XRPLF/rippled/pull/4849 

**NOTE** This PR should be only merged by someone with administrative access to the repo, since it makes use of `secrets.CODECOV_TOKEN`. If I understand github workflows correctly, a commit (merge) made by someone _without_ access to repo secrets would result in this change _not_ having this access either, which will lead to [spurious failures of codecov integration](https://community.codecov.com/t/upload-issues-unable-to-locate-build-via-github-actions-api/3954)

### CODE REVIEW

**Review by cindyyan317 — DISMISSED** [review]
👍

File: .codecov.yml [hunk 1]
```diff
@@ -0,0 +1,7 @@
+coverage:
+  status:
+    project:
+      default:
+        target: 50%
```
Comment (godexsoft) [inline .codecov.yml:5]: Is the plan to gradually update this value to a higher one as we add more coverage?

**Review by godexsoft — DISMISSED** [review]
Very nice! Thank you for fixing codecov report 🚀 🥇

File: .codecov.yml (hunk 1 above)
Comment (Bronek) [reply .codecov.yml:5]: Yes that's definitely the plan ! I will leave it to you :)

**Review by cindyyan317 — APPROVED** [review]

**Pull request merged**